### ISSUE TITLE
Don't require fusermount when run as root or in root-mapped unprivileged user namespace

### ISSUE THREAD

**Issue by DrDaveD** [rfjakob/gocryptfs#749]
Most FUSE packages are built on fuse-libs, and that does not use fusermount when run as root or in the pseudo-root of a root mapped unprivileged user namespace (`unshare -rm`).  That is not the case for gocryptfs; it needs fusermount anyway, even though it does not need the setuid-root functionality.  That is a problem because it requires installing the `fuse` package, which has a setuid-root fusermount in it, and some system administrators don't like taking the risk of setuid-root programs.

In particular, gocryptfs is [now](https://github.com/apptainer/apptainer/pull/1359) being packaged with [apptainer](https://github.com/apptainer/apptainer), and this behavior is requiring adding the `fuse` package as a dependency.

**Comment by rfjakob** [comment]
Hi, please see https://github.com/rfjakob/gocryptfs/issues/697

Tldr: it's done

**Comment by DrDaveD** [comment]
Great!  I tried to search for it but I bet I only looked at open issues.  Will it be tagged into a release soon?

**Comment by DrDaveD** [comment]
@rfjakob Could you please tag a new release with this behavior change?  It would make things a lot simpler for me.

**Comment by rfjakob** [comment]
@DrDaveD done: https://github.com/rfjakob/gocryptfs/releases/tag/v2.4.0

**Comment by DrDaveD** [comment]
Thanks!
